frame/assets/src/lib.rs

```diff
@@ -1511,6 +1511,45 @@ pub mod pallet {
 			let id: T::AssetId = id.into();
 			Self::do_refund(id, ensure_signed(origin)?, allow_burn)
 		}
+
+		/// Sets the minimum balance of an asset. Only works if there aren't any
+		/// accounts that are holding the asset or if the new vlaue of new_balance
+		/// is less than the old one.
+		///
+		/// Origin must be Signed and the sender has to be the Owner of the
+		/// asset `id`.
+		///
+		/// - `id`: The identifier of the asset.
+		/// - `min_balance`: The new value of `min_balance`.
+		///
+		/// Emits `AssetStatusChanged` event when successful.
+		#[pallet::call_index(28)]
+		#[pallet::weight(10_000)]
+		pub fn set_min_balance(
+			origin: OriginFor<T>,
+			id: T::AssetIdParameter,
+			min_balance: T::Balance,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let id: T::AssetId = id.into();
+
+			let mut details = Asset::<T, I>::get(id).ok_or(Error::<T, I>::Unknown)?;
+			ensure!(origin == details.owner, Error::<T, I>::NoPermission);
+
+			let old_min_balance = details.min_balance;
+			// Ensure that either the new min_balance is less than old min_balance or there aren't
+			// any accounts holding the asset.
+			ensure!(
+				min_balance < old_min_balance || details.accounts == 0,
+				Error::<T, I>::NoPermission
+			);
+
+			details.min_balance = min_balance;
+			Asset::<T, I>::insert(&id, details);
+
+			Self::deposit_event(Event::AssetStatusChanged { asset_id: id });
```

Comment:
Shall we emit smth more specific? Like AssetMinBalanceChanged and pass the new value into the event data?

Reply:
Makes sense, I will add that.